From 3a210b9477e7ae3ab5619941bbec38a06e54a175 Mon Sep 17 00:00:00 2001
From: Lu Chongzhi <chongzhi_lui@c-sky.com>
Date: Thu, 20 Jun 2019 19:57:45 +0800
Subject: [PATCH 06/10] thead: Fixup relocaton R_RISCV_ALIGN with our own
 compiler

 - Return 0 even relocation type is R_RISCV_ALIGN, the reason is:
   When call get_user/put_user, compiler will insert ".balign 4" in section fixup,
   then the ko file contains R_RISCV_ALIGN relocation flag, which can not be handled
   by insmod process.

Signed-off-by: Lu Chongzhi <chongzhi_lu@c-sky.com>
Signed-off-by: Guo Ren <guoren@linux.alibaba.com>
---
 arch/riscv/kernel/module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/kernel/module.c b/arch/riscv/kernel/module.c
index 6bf5b16..483d5f0 100644
--- a/arch/riscv/kernel/module.c
+++ b/arch/riscv/kernel/module.c
@@ -253,7 +253,7 @@ static int apply_r_riscv_align_rela(struct module *me, u32 *location,
 	pr_err(
 	  "%s: The unexpected relocation type 'R_RISCV_ALIGN' from PC = %p\n",
 	  me->name, location);
-	return -EINVAL;
+	return 0; /* Do not return -EINVAL when relocation type is R_RISCV_ALIGN */
 }
 
 static int apply_r_riscv_add32_rela(struct module *me, u32 *location,
-- 
2.7.4

